File: lib/core/shared_data_source/websocket/msg_buffer.dart
import 'package:synchronized/synchronized.dart';
import 'msg_model.dart';

class MsgBuffer {
  final Lock _lock = Lock();
  Map<String, MsgModel> buffer = {};
  void saveCopy(MsgModel msgModel) async {
    await _lock.synchronized(() => buffer[msgModel.id] = msgModel);
  }

  void deleteCopy(String id) async {
    await _lock.synchronized(() => buffer.remove(id));
  }

  Future<List<MsgModel>> getBufferContent() async {
    List<MsgModel> msgs = [];
    await _lock.synchronized(() => buffer.forEach((_, value) {
          msgs.add(value);
        }));
    return msgs;
  }
}

File: src/websocket/msg_buffer.ts
import { MsgModel } from "./msg_model";

export class MsgBuffer {

    private _buffer: Map<string, MsgModel> = new Map()
    public saveCopy(msgModel: MsgModel): void {
        this._buffer.set(msgModel.id, msgModel)
    }

    public deleteCopy(id: string): void {
        this._buffer.delete(id)
    }


    public getBufferContent(): Array<MsgModel> {
        return Array.from(this._buffer.values())
    }
}
